pkg/analysis/passes/version/version.go

```diff
@@ -98,18 +102,12 @@ func getPluginDataFromGrafanaCom(pluginId string) (*PluginStatus, error) {
 
 	// != 200 = something went wrong. We can't check the plugin
 	if response.StatusCode != http.StatusOK {
-		return nil, err
-	}
-
-	content, err := ioutil.ReadAll(response.Body)
-	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("wrong status code, expected 200 got %d", response.StatusCode)
 	}
 
 	status := PluginStatus{}
-
-	err = json.Unmarshal(content, &status)
-	if err != nil {
+	defer response.Body.Close()
```

Comment:
My bad, sorry this `Close` should go above, right after the `if err != nil` block on line 94 (so on line 97). This is because we are returning early if `response.StatusCode != http.StatusOK` or `response.StatusCode == http.StatusNotFound`, and in that case we would not close the response body

Reply:
makes sense!